### DISCUSSION TITLE
Read-Only text fields on the account console (v2)

### DISCUSSION THREAD

**Discussion by FlxPeters** [keycloak/keycloak#21944] [Q&A]
Hey fellow community,

we have our user data stored in an LDAP directory attached to Keycloak. This works like charm.
We also have the declarative user profile feature enabled we want to make all fields read only in the users account console except the language setting and password reset.

For now I didn't find a way to configure the account console fields like `firstName`or `lastName` with attributes to make them read-only.

So how can i disable this fields in order to prevent the user trying to edit this values?

Thanks!

**Comment by darius-m** [comment]
The only solution I could find that is somewhat easy to implement for the front-end was to create a theme that is based on Keycloak v2 and add the `disabled` attribute to the fields. However, this does *not* prevent the users from changing their forms in HTML and removing the `disabled` attribute to change the fields.

To actually prevent changes you will need to use a server-side feature (e.g. if the LDAP binding is read-only, the changes should fail by default). I cannot say how you could prevent changes server-side; the declarative profile may be of some use, but I have not used them, so I am not familiar with this functionality.

**Reply by FlxPeters** [reply]
Thanks for your feedback. It looks like the new account console v3 should implement this. Not tested yet, but it's in the release notes.
